<?xml version="1.0" ?>
<launch>
    <arg name="eye_on_hand" default="false" />
    <arg name="namespace_prefix" default="panda_eih_calib" if="$(arg eye_on_hand)" /> 
    <arg name="namespace_prefix" default="panda_eob_calib" unless="$(arg eye_on_hand)"/>
    <arg name="robot_ip" />

    <!-- <include file="$(find easy_handeye)/launch/publish.launch">
        <arg name="eye_on_hand" value="$(arg eye_on_hand)" />
        <arg name="namespace_prefix" value="$(arg namespace_prefix)" />
    </include> -->

    <node name="virtual_joint_point_boardcaster_1" pkg="tf2_ros" type="static_transform_publisher" args="0.8073418735153907 0.08408076947494328 0.4628911890570747 0.6959747938926792 0.68032001124097 -0.165550968752189 -0.1593004874956155 world camera_link"/>

    <node name="virtual_joint_point_boardcaster_2" pkg="tf2_ros" type="static_transform_publisher" args="0 0 0 0 0 0 world panda_link0"/>
    
    <!-- (start your robot's MoveIt! stack, e.g. include its moveit_planning_execution.launch) -->
    <include file="$(find panda_moveit_config)/launch/panda_control_moveit_rviz.launch">
        <arg name="robot_ip" value="$(arg robot_ip)"/> <!-- set your robot ip -->
        <arg name="load_gripper" value="true"/>
        <arg name="launch_rviz" value="true"/>
    </include>        
    
    <!-- (start your tracking system's ROS driver) -->
    <include file="$(find realsense2_camera)/launch/rs_camera.launch"> </include>     

    <!-- fill in the following parameters according to your tracking system -->
    <arg name="markerId"        default="7"/>      <!-- set your marker id -->
    <!--arg name="markerSize"      default="0.037"/!-->    <!-- in m -->
    <arg name="markerSize"      default="0.04"/>
    <arg name="eye"             default="left"/>
    <arg name="marker_frame"    default="aruco_marker_frame"/> 
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->    
    <arg name="camera_frame" default="camera_color_frame" />
    <arg name="camera_image_topic" default="/camera/color/image_raw" />
    <arg name="camera_info_topic" default="/camera/color/camera_info" /> 

    <node pkg="aruco_ros" type="single" name="aruco_single">
        <remap to="$(arg camera_info_topic)" from="/camera_info" />
        <remap to="$(arg camera_image_topic)" from="/image" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <param name="camera_frame"       value="$(arg camera_frame)"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

</launch>
